### ISSUE TITLE
[Request] Please add Snap tracks for major versions or at least LTS versions

### ISSUE THREAD

**Issue by grantzvolsky** [dotnet/core#3468]
## Problem description
While I very much appreciate that .NET Core is available as a [snap package](https://snapcraft.io/dotnet-sdk), I find it unfortunate that there are only three available release channels, neither of which allows the user to stick to a certain major version. Instead the lack of channels forces the user to always update to the latest major version. This is particularly inconvenient when frequently switching between projects that use different .NET Core versions.

## Desired state
There is a [track](https://snapcraft.io/docs/channels#heading--tracks) for all LTS releases if not for all major versions.

**Comment by jamshedd** [comment]
@leecow, @vivmishra

**Comment by leecow** [comment]
Thanks for that feedback @grantzvolsky. We're still working to mature the Snap offering and this will help. One thing to keep in mind is that the SDK is not part of the LTS definition. That only refers to the Runtime. That said, we may be able to factor things in a way that makes sense. 

Something along these lines?

* dotnet-sdk : always the latest
* dotnet-sdk-[major.minor] : latest patch release for channel
* dotnet-sdk-lts : while not an lts sdk, it would be the latest sdk patch version which aligns with the LTS runtime.

**Comment by grantzvolsky** [comment]
I understand that you maintain one snap per runtime major version. This allowed you to have multiple runtime versions before parallel installs for confined snaps were supported. At present, with support for parallel confined snaps, it should be possible to have just one snap for all runtime channels.

The sdk is different in that it is a 'classic' snap, and parallel installs of classic snaps are not yet supported [[1]](https://forum.snapcraft.io/t/supporting-parallel-installs-on-desktop-also-classic/11760/2) [[2]](https://forum.snapcraft.io/t/godot-engine-and-runtime-snap-naming/11774). I would still suggest using a single snap with multiple tracks. Using multiple sdk versions in parallel won't be possible for now, but one will be able to switch between tracks seamlessly with `snap refresh dotnet-sdk --channel 2.1/stable`. An example snap that does this is [kubectl](https://snapcraft.io/kubectl). 

The channels may then be:
* latest/stable
* latest/candidate
* latest/beta
* latest/edge
* 3.0/stable
* 3.0/candidate
* 3.0/beta
* 3.0/edge
* 2.2/stable
* 2.2/candidate
* 2.2/beta
* 2.2/edge
* etc.

**Comment by leecow** [comment]
Great information - thanks again! I'm out of town at the moment but will meet with the team next week to discuss this in detail. @MichaelSimons @dleeapho

**Comment by leecow** [comment]
Update - Snap Store folks should have our Tracks set up in the next day or so. https://forum.snapcraft.io/t/request-tracks-for-dotnet-runtime-and-dotnet-sdk/13469/4

**Comment by carlossanlop** [comment]
@leecow did this get resolved?

**Comment by leecow** [comment]
Yes, major.minor tracks are implemented. Still in a discussion concerning LTS.
